File: Java/LinkedLists/src/Cell.java
public class Cell <T>{

    private T element;
    private Cell<T> next;

    @Override
    public String toString(){
        return "Element: " + this.element + " | " + "Next: " + this.next;
    }
    public Cell(T element){
        this.element = element;
        this.next = null;
    }

    public Cell(T element, Cell<T> next) {
        this.element = element;
        this.next = next;
    }

    public void setElement(T element){
        this.element = element;
    }

    public T getElement(){
        return this.element;
    }

    public void setNext(Cell<T> next){
        this.next = next;
    }

    public Cell<T> getNext() {
        return this.next;
    }
}

File: Kotlin/LinkedLists/src/Cell.kt
class Cell<T>(private var element: T, private var constNext: Cell<T>?) {
    private var next: Cell<T>? = constNext

    constructor(element: T): this(element, constNext = null)



    fun getElement(): T {
        return this.element
    }

    fun setElement(element: T?){
        this.element = null as T
    }

    fun setNext(next: Cell<T>?) {
        this.next = next
    }

    fun getNext() : Cell<T>? {
        return this.next
    }

    override fun toString(): String {
        return "Element: ${this.element} | Next: ${this.next}"
    }


}